File: C_Sharp/Framework.Tests/CustomerServices/Customer.cs
using System.Collections.Generic;
using Framework.Data_Manipulation;
using Framework.Domain;

namespace Framework.Tests.CustomerServices
{
    public class Customer : DomainObject
    {
        public class InnerClass { }

        public Customer(IBaseMapper mapper) : base(mapper)
        {
        }

        [IdentityField]
        public string Number { get; set; }

        public string Name { get; set; }

        [IgnorePropertyMonitoring]
        public bool NotSoHelpfulMember { get; set; }

        public InnerClass Temp { get; set; }
    }
}

File: Android/app/src/test/java/com/CustomerServices/Customer.java
package com.CustomerServices;

import com.IdentityFieldAnnotation;
import com.DataManipulation.BaseMapperInterfaces.IBaseMapper;
import com.Domain.DomainObject;

/**
 * Created by aiko on 3/5/17.
 */

public class Customer extends DomainObject {
    public Customer(IBaseMapper mapper)
    {
        super(mapper);
    }

    public Customer(IBaseMapper mapper, String number)
    {
        this(mapper);

        Number = number;
    }

    @IdentityFieldAnnotation
    public String Number;

    public String Name;
}
